README.md
=== Genesis Simple Footer Widgets ===
Contributors: devdesignstampa, hellofromTonya
Donate link: http://wpdevelopersclub.com/
Tags: genesis, genesiswp, widgets, footer widgets, extra widgets
Requires at least: 3.5
Tested up to: 4.6.1
Stable tag: 1.2
License: GPLv2 or later
License URI: http://www.gnu.org/licenses/gpl-2.0.html


A simple way to change how many footer widgets your site uses.


## Description

Easily go from four widgets to three or three to six from a dropdown in Genesis Theme Settings.

Genesis Simple Footer Widgets lets you go from one footer widget to six or no footer widgets to six in seconds without hiring a developer.


[Source code on GitHub](https://github.com/joe-dooley/genesis-simple-footer-widgets/)


#### Special thanks to:
* C. Paul for inspiring me to build the plugin with his article 
[Controlling Footer Widget Columns from Theme Settings Page in Genesis](http://genesisdeveloper.mecontrolling-footer-widget-columns-from-theme-settings-page-in-genesis/) .
* [Cathi Bosco of C&D Studios](http://www.canddstudios.com/) , for designing the banner image for Genesis Simple Footer Widgets.



## Support

* This plugin works on all Studiopress themes and most well coded Genesis child themes.
* Please use the plugins [support forum](https://wordpress.org/support/plugin/genesis-variable-footer-widgets) for help. Don't ask for help in a review.
* [Contact me](https://www.developingdesigns.com/contact-us/) for translations.



## Installation

1. Upload the folder to the `/wp-content/plugins/` directory
2. Activate the plugin through the `Plugins` menu in WordPress



## How To Use

1. Navigate to the `Genesis > Theme Settings` menu
2. Select how many widgets you need from the dropdown and click save.
3. Go to `Appearance > Widgets` and add new widgets to the new widget areas.



## Frequently Asked Questions

**Is the CSS responsive?**

Yes, the widgets are fully responsive. For example, if you have 6 widgets selected. You will see 6 widgets in one row on desktop, 3 widgets divided into
two rows on tablets and on mobile devices you will have all of your widgets stacked on top of each other.



== Screenshots ==

1. Select number of widgets from a dropdown in `Genesis Theme Settings`
2. One widget
3. Two widgets
4. Three widgets
5. Four widgets
6. Five widgets
7. Six widgets
8. Remove footer widgets by selecting zero widgets



CHANGELOG.md
1.2 (2016-10-17)
- Added support for translations via load_plugin_textdomain()
- Added conditionals to make sure plugin cannot be activated on
  non genesis themes and anyone using PHP 5.3.x


1.1 (2016-10-16)
- Added a high-DPI banner designed by [Cathi Bosco](http://www.canddstudios.com/)
- Version bump
- Code refactoring
- Minified CSS

1.0.3 (2015-07-08)
- Updated readme.txt and README.md

1.0.2 (2015-07-08)
- Changed Plugin name to Genesis Simple Footer Widgets
 
1.0.1 (2015-07-08)
- Added CHANGELOG.md
- Removed unnecessary files.
- Edited readme.txt and README.md
- Added instructions within Genesis Theme Settings 

1.0.0 (2015-05-26)
- Initial release.


plugin.php
<?php
/**
 * Plugin Name:  Genesis Simple Footer Widgets
 * Plugin URI:   https://github.com/joedooley/genesis-simple-footer-widgets
 * Description:  A simple way to change how many footer widgets your site uses. Easily go
 *               from four widgets to three or three to six from a dropdown in Genesis Theme Settings.
 * Author:       Joe Dooley
 * Author URI:   http://www.developingdesigns.com/
 * Version:      1.2
 * Text Domain:  genesis-simple-footer-widgets
 * Domain Path:  languages
 * Requires PHP: 5.4
 *
 * Genesis Simple Footer Widgets is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 2 of the License, or
 * any later version.
 *
 * Genesis Simple Footer Widgets is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Genesis Title Toggle. If not, see <http://www.gnu.org/licenses/>.
 *
 * @package    JD_Genesis_Simple_Footer_Widgets
 * @author     Joe Dooley <hello@developingdesigns.com>
 * @license    GPL-2.0+
 * @copyright  2015 Joe Dooley, Developing Designs
 */

namespace JD_Genesis_Simple_Footer_Widgets;

/**
 * If this file is called directly, abort.
 */
if ( ! defined( 'WPINC' ) ) {
	die;
}



add_action( 'init', __NAMESPACE__ . '\load_textdomain' );
/**
 * Loads plugins translated strings. Were loading load_plugin_textdomain()
 * on 'init' hook.
 *
 * @since 1.2
 */
function load_textdomain() {
	load_plugin_textdomain( 'genesis-simple-footer-widgets', false, basename( dirname( __DIR__ ) ) . '/languages/' );
}


add_action( 'admin_notices', __NAMESPACE__ . '\genesis_check' );
/**
 * Display admin notice if Genesis isn't installed or 'PHP version < 5.4'.
 *
 * @since 1.2
 */
function genesis_check() {
	if ( 'genesis' !== basename( TEMPLATEPATH ) ) {
		deactivate_plugins( plugin_basename( __FILE__ ) );
		wp_die( __( 'Sorry, Genesis Simple Footer Widgets needs the Genesis theme to be activated in order to function properly. Please install Genesis and activate the theme.', 'genesis-simple-footer-widgets' ) );
	}
}



add_action( 'admin_notices', __NAMESPACE__ . '\min_php_check' );
/**
 * Do not activate plugin if PHP < 5.4
 *
 * @param $version int PHP version
 * @since 1.2
 */
function min_php_check( $version ) {
	$min_php_version = 5.4;

	if ( version_compare( $min_php_version, $version, '<' ) ) {
		deactivate_plugins( plugin_basename( __FILE__ ) );
		wp_die( sprintf( __( 'Sorry, Genesis Simple Footer Widgets requires PHP version %1$d or greater to function properly. You have PHP version %2$d. Please contact your hosting provider for assistance.', 'genesis-simple-footer-widgets' ) ) );
	}
}



add_action( 'genesis_setup', __NAMESPACE__ . '\launch', 20 );
/**
 * Launch this plugin once Genesis fires up
 *
 * @since 1.0.2
 */
function launch() {

	$path = plugin_dir_path( __FILE__ );
	require_once $path . 'class-plugin.php';

	$defaults = [
		'views'         => [
			'metabox'   => $path . 'lib/views/settings-box.php',
		],
		'default_number_of_widgets' => 3,
	];

	$config = apply_filters( 'gsfw_configuration_parameters', $defaults );

	new Genesis_Simple_Footer_Widgets( $config, $path, $defaults );
}


lib/admin/class-metabox.php
<?php
/**
 * Genesis Simple Footer Widgets
 *
 * @package     JD_Genesis_Simple_Footer_Widgets
 * @since       1.0.2
 * @author      Joe Dooley <hello@developingdesigns.com>
 * @link        https://github.com/joedooley/genesis-simple-footer-widgets
 * @license     GNU General Public License 2.0+
 * @copyright   2015 Joe Dooley, Developing Designs
 */

namespace JD_Genesis_Simple_Footer_Widgets\Admin;

class Metabox {

	protected $config = array();

	/****************************
	 * Instantiate & Initialize
	 ***************************/

	/**
	 * Instantiate the plugin
	 *
	 * @since 1.0.2
	 *
	 * @param array     $config         Configuration parameters
	 */
	public function __construct( array $config ) {
		$this->config = $config;
		$this->init_hooks();
	}

	/**
	 * Initialize the plugin's properties
	 *
	 * @since 1.0.2
	 *
	 * @return null
	 */
	protected function init_hooks() {
		add_action( 'genesis_theme_settings_metaboxes', array( $this, 'register_settings_box' ) );
	}

	/*****************************
	 * Public Methods & Callbacks
	 ****************************/

	/**
	 * Register additional metaboxes to Genesis > Theme Settings
	 *
	 * @param string    $pagehook      $_genesis_theme_settings_pagehook
	 * @return null
	 */
	public function register_settings_box( $pagehook ) {
		add_meta_box(
			'jd-gsfw-settings',
			'Footer Widgets',
			[
				$this,
				'render_metabox',
			],
			$pagehook,
			'main',
			'high'
		);
	}

	/**
	 * Render the Metabox
	 *
	 * @since 1.0.2
	 *
	 * @return null
	 */
	public function render_metabox() {
		$number_of_widgets = genesis_get_option( 'footer_widgets' );
		$number_of_widgets = is_numeric( $number_of_widgets ) ? (int) $number_of_widgets : 0;

		if ( is_readable( $this->config['views']['metabox'] ) ) {
			include $this->config['views']['metabox'];
		}
	}
}


class-plugin.php
<?php
/**
 * Genesis Simple Footer Widgets
 *
 * @package     JD_Genesis_Simple_Footer_Widgets
 * @since       1.0.2
 * @author      Joe Dooley <hello@developingdesigns.com>
 * @link        https://github.com/joedooley/genesis-simple-footer-widgets
 * @license     GNU General Public License 2.0+
 * @copyright   2015 Joe Dooley, Developing Designs
 */

namespace JD_Genesis_Simple_Footer_Widgets;

use JD_Genesis_Simple_Footer_Widgets\Admin\Metabox;

// Oh no you don't. Exit if accessed directly
if ( ! defined( 'ABSPATH' ) ) {
	exit( 'Cheatin&#8217; uh?' );
}

class Genesis_Simple_Footer_Widgets {

	/**
	 * Number of footer widgets
	 *
	 * @var int
	 */
	protected $number_of_footer_widgets = 3;

	/**
	 * The plugin's version
	 *
	 * @var string
	 */
	const VERSION = '1.1';

	/**
	 * The plugin's minimum WordPress requirement
	 *
	 * @var string
	 */
	const MIN_WP_VERSION = '3.5';

	/**
	 * Configuration parameters
	 *
	 * @var array
	 */
	protected $config = array();

	/**
	 * Plugin's dir path
	 *
	 * @var string
	 */
	protected $plugin_dir = '';

	/**
	 * Plugin's URL path
	 *
	 * @var string
	 */
	protected $plugin_url = '';

	/**
	 * Instance of Metabox
	 *
	 * @var
	 */
	protected $metabox;

	/*************************
	 * Getters
	 ************************/

	public function version() {
		return self::VERSION;
	}

	public function min_wp_version() {
		return self::MIN_WP_VERSION;
	}

	/****************************
	 * Instantiate & Initialize
	 ***************************/

	/**
	 * Instantiate the plugin
	 *
	 * @since 1.0.2
	 *
	 * @param array  $config     Configuration parameters
	 * @param string $plugin_dir Plugin directory
	 * @param array  $defaults   Default configuration parameters
	 */
	public function __construct( array $config, $plugin_dir, array $defaults ) {
		$this->init_properties( $config, $plugin_dir, $defaults );
		$this->init_hooks();
	}

	/**
	 * Initialize the plugin's properties
	 *
	 * @since 1.0.2
	 *
	 * @param array     $config         Configuration parameters
	 * @param string    $plugin_dir     Plugin dir
	 * @param array     $defaults       Default configuration parameters
	 */
	protected function init_properties( array $config, $plugin_dir, array $defaults ) {
		$this->config = wp_parse_args( $config, $defaults );
		$this->plugin_dir = $plugin_dir;

		$this->plugin_url = plugin_dir_url( __FILE__ );
		if ( is_ssl() ) {
			$this->plugin_url = str_replace( 'http://', 'https://', $this->plugin_url );
		}

		$this->number_of_footer_widgets = (int) genesis_get_option( 'footer_widgets' );
		$this->number_of_footer_widgets = $this->number_of_footer_widgets > -1 ? $this->number_of_footer_widgets : 3;
	}

	/**
	 * Initialize hooks
	 *
	 * @since 1.0.2
	 */
	protected function init_hooks() {

		add_action( 'after_setup_theme',                array( $this, 'add_theme_support' ), 9 );

		if ( $this->number_of_footer_widgets > 0 ) {
			add_filter( 'genesis_attr_footer-widgets',  array( $this, 'footer_widgets_attr' ) );
		}

		add_action( 'wp_enqueue_scripts',               array( $this, 'enqueue' ) );
		add_filter( 'genesis_theme_settings_defaults',  array( $this, 'set_defaults' ) );
		add_action( 'genesis_settings_sanitizer_init',  array( $this, 'sanitize' ) );
		add_action( 'admin_init',                       array( $this, 'init_admin' ) );
	}

	/*****************************
	 * Public Methods & Callbacks
	 ****************************/

	/**
	 * Initialize the Metabox in the admin area
	 *
	 * @since 1.0.2
	 */
	public function init_admin() {
		include $this->plugin_dir . 'lib/admin/class-metabox.php';

		$this->metabox = new Metabox( $this->config );
	}

	/**
	 * Add theme support for the footer widgets
	 *
	 * Note: We hook into after_setup_theme before Genesis calls genesis_register_footer_widget_areas()
	 * to ensure we are overriding the Child theme's setting for footer widgets
	 *
	 * @since 1.0.2
	 */
	public function add_theme_support() {
		add_theme_support( 'genesis-footer-widgets', $this->number_of_footer_widgets );
	}

	/**
	 * Register the styles
	 *
	 * @since 1.0.2
	 */
	public function enqueue() {
		wp_enqueue_style(
			'gsfw-stylesheet',
			$this->plugin_url . 'css/style.min.css',
			false,
			self::VERSION
		);
	}

	/**
	 * Register Defaults
	 *
	 * @author Bill Erickson
	 * @link http://www.billerickson.net/genesis-theme-options/
	 *
	 * @param array     $defaults
	 * @return array    modified defaults
	 */
	public function set_defaults( $defaults ) {
		$defaults['footer_widgets'] = '3';

		return $defaults;
	}

	/**
	 * Add css class attributes
	 *
	 * @since 1.0.2
	 *
	 * @param array     $attributes
	 * @return array
	 */
	public function footer_widgets_attr( $attributes ) {
		$attributes['class'] .= ' gsfw-footer-widgets-' . genesis_get_option( 'footer_widgets' );

		return $attributes;
	}

	/**
	 * Sanitization
	 *
	 * Register our new option values with the no_html sanitization type defined within Genesis.
	 *
	 * @author Bill Erickson
	 * @link http://www.billerickson.net/genesis-theme-options/
	 *
	 */
	public function sanitize() {
		genesis_add_option_filter(
			'no_html',
			GENESIS_SETTINGS_FIELD,
			[
				'footer_widgets',
			]
		);
	}
}
